### PR TITLE
created components, redux module, helpers, screen related to filtering

### PR DESCRIPTION
**Pull request by science-girl**
refactored ListingList to Card so it can be reused for Filtering

property type buttons in redux

refactored buttons into a component

refactored search results with redux so filter can be aware of listings and location

filtering mostly done. just need slider and number picker

built component for inputing bedroom and bathroom numbers

refactored with a partially applied function to reduce clutter in filtering files

added multislider

refactored slider

added comments

fixed merge conflicts

deleted env

### CODE REVIEW

File: js/components/IncrementalInput/IncrementalInput.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const IncrementalInput = ({ label, value, incrementValue, decrementValue }) => {
+  return (
+    <View style={styles.container}>
+      <View style={styles.label}>
+        <Text>{label}</Text>
+      </View>
+      <View style={styles.incrementContainer}>
+        <TouchableOpacity onPress={() => decrementValue(true)}>
+          {' '}
```
Comment (siddharthparmar7) [inline js/components/IncrementalInput/IncrementalInput.js]: Please remove {'  '}.

File: js/components/IncrementalInput/IncrementalInput.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          <Text style={styles.icon}>{'<'}</Text>
+        </TouchableOpacity>
+        <Text style={styles.value}>{value}</Text>
+        <TouchableOpacity onPress={() => incrementValue(false)}>
+          {' '}
```
Comment (siddharthparmar7) [inline js/components/IncrementalInput/IncrementalInput.js]: Please remove {'  '}.

File: js/components/IncrementalInput/styles.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { StyleSheet } from 'react-native';
+import { colors /*typography*/ } from '../../config/styles';
```
Comment (siddharthparmar7) [inline js/components/IncrementalInput/styles.js:2]: Please remove commented code.

File: js/components/IncrementalInput/styles.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    width: 240,
+    height: 40,
+    borderRadius: 5,
+    flex: 1,
+    flexDirection: 'row'
+  },
+  label: {
+    marginTop: 10,
+    marginLeft: 5
+    //fontFamily: typography.OS_REGULAR
```
Comment (siddharthparmar7) [inline js/components/IncrementalInput/styles.js:18]: Please remove this comment.

File: js/components/IncrementalInput/styles.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  icon: {
+    fontSize: 20
+    //fontFamily: typography.OS_REGULAR
```
Comment (siddharthparmar7) [inline js/components/IncrementalInput/styles.js:22]: Please remove this comment.

File: js/components/IncrementalInput/styles.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  value: {
+    fontSize: 14,
+    marginTop: 5
+    //fontFamily: typography.OS_REGULAR
```
Comment (siddharthparmar7) [inline js/components/IncrementalInput/styles.js:27]: Please remove this comment.

File: js/components/ListItem/styles.js [hunk 7]
```diff
@@ -16,7 +16,7 @@ export const styles = StyleSheet.create({
     marginTop: 15,
     marginBottom: 10
   },
-  listing: { marginTop: 15 },
+  listing: { marginTop: 5 },
```
Comment (siddharthparmar7) [inline js/components/ListItem/styles.js:19]: Please remove comments from line 2, 22, 30, 36 and 38.

File: js/components/OutlineButton/OutlineButton.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  return isPressed
+    ? button(label, styles.pressedButton, styles.pressedText, onPress)
+    : button(label, styles.button, styles.text, onPress, disabled);
+};
+
+OutlineButton.propTypes = {
+  label: PropTypes.string.isRequired,
+  isPressed: PropTypes.bool.isRequired,
+  onPress: PropTypes.func.isRequired,
+  disabled: PropTypes.bool
```
Comment (siddharthparmar7) [inline js/components/OutlineButton/OutlineButton.js:24]: Please add a default prop if you are not using .isRequired.

File: js/components/OutlineButton/styles.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import { StyleSheet } from 'react-native';
+import { colors /*typography*/ } from '../../config/styles';
```
Comment (siddharthparmar7) [inline js/components/OutlineButton/styles.js:2]: Please remove this comment.

File: js/components/OutlineButton/styles.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    width: 100,
+    height: 30,
+    borderRadius: 5,
+    justifyContent: 'center',
+    alignItems: 'center',
+    margin: 2
+  },
+  text: {
+    color: colors.night
+    //fontFamily: typography.OS_REGULAR
```
Comment (siddharthparmar7) [inline js/components/OutlineButton/styles.js:29]: Please remove this comment.

File: js/components/OutlineButton/styles.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  pressedText: {
+    color: colors.BACKGROUND
+    //fontFamily: typography.OS_REGULAR
```
Comment (siddharthparmar7) [inline js/components/OutlineButton/styles.js:33]: Please remove this comment.

File: js/components/Slider/styles.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import { StyleSheet } from 'react-native';
+import { colors /*typography*/ } from '../../config/styles';
```
Comment (siddharthparmar7) [inline js/components/Slider/styles.js:2]: Please remove this comment

File: js/components/Slider/styles.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+export const styles = StyleSheet.create({
+  label: {
+    marginBottom: 20
+    //fontFamily: typography.OS_REGULAR
```
Comment (siddharthparmar7) [inline js/components/Slider/styles.js:7]: Please remove this comment

File: js/screens/Filtering/FilteringScreen.js [hunk 14]
```diff
@@ -0,0 +1,177 @@
+import React, { Component } from 'react';
```
Comment (siddharthparmar7) [inline js/screens/Filtering/FilteringScreen.js:1]: Please rename this file to FilteringContainer.js

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please delete .env file.

Nice work, thank you!

Please rename your file FavouriteScreen.js to FavoriteContainer.js in the screens dir.

Thank you!

**Comment by science-girl** [comment]
We are in the midsts of re-naming all of our containers to screens - that is in Nicholas' PR, so I pre-emptively named mine FilteringScreen instead of Container.

The commented out typography and fontFamily fields are there because the linking is also in Nicholas' PR, so once we have that merged, we can uncomment them.

**Review by siddharthparmar7 — APPROVED** [review]

**Pull request merged**